File: week1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Asks a height from the user that is between 1 and 8
    do
    {
        height = get_int("What is the height? ");
    }
    while (height < 1 || height > 8);

    // outer loop for printing new lines
    for (int i = 1; i <= height; i++)
    {
        // loop for printing the spaces
        // they are always equal the current index - 1
        for (int j = 1; j <= height - i; j++)
        {
            printf(" ");
        }

        // loop for printing the hashes
        for (int k = 1; k <= i; k++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: week6/sentimental-mario-more/mario.py
def main():

    h = get_int("Height: ")

    mario(h)


def mario(height):
    # loop for printing the new lines
    for i in range(1, height + 1):

        # loop for printing the set of spaces
        for j in range(height - i):
            print(" ", end="")

        # loop for printing the 1st pyramid
        for k in range(i):
            print("#", end="")

        # loop for printing the blank space between the pyramids
        for m in range(2):
            print(" ", end="")

        # loop for printing the 2nd pyramid
        for n in range(i):
            print("#", end="")

        print()


def get_int(prompt):
    while True:
        try:
            h = int(input(prompt))

            if h >= 1 and h <= 8:
                return h
        except EOFError:
            break
        except:
            pass


if __name__ == "__main__":
    main()